docker/Makefile

```diff
@@ -0,0 +1,51 @@
+NAMESPACE ?= puppet
+git_describe := $(shell git describe)
+vcs_ref := $(shell git rev-parse HEAD)
+build_date := $(shell date -u +%FT%T)
+hadolint_available := $(shell hadolint --help > /dev/null 2>&1; echo $$?)
+hadolint_command := hadolint --ignore DL3008 --ignore DL3018 --ignore DL3028 --ignore DL4000 --ignore DL4001
+hadolint_container := hadolint/hadolint:latest
+pwd := $(shell pwd)
+export BUNDLE_PATH = $(pwd)/.bundle/gems
+export BUNDLE_BIN = $(pwd)/.bundle/bin
+export GEMFILE = $(pwd)/Gemfile
+
+version := $(shell echo $(git_describe) | sed 's/-.*//')
+dockerfile := Dockerfile
+
+prep:
+	@git fetch --unshallow 2> /dev/null ||:
+	@git fetch origin 'refs/tags/*:refs/tags/*'
+
+lint:
+ifeq ($(hadolint_available),0)
+	@$(hadolint_command) ezbake/$(dockerfile)
+else
+	@docker pull $(hadolint_container)
+	@docker run --rm -v $(PWD)/ezbake/$(dockerfile):/Dockerfile -i $(hadolint_container) $(hadolint_command) Dockerfile
+endif
+
+build: prep
+	  @docker build \
+			--pull \
+			--build-arg vcs_ref=$(vcs_ref) \
+			--build-arg build_date=$(build_date) \
+			--build-arg version=$(version) \
+			--file ezbake/$(dockerfile) \
+			--tag $(NAMESPACE)/ezbake:$(version) $(pwd)/..
+ifeq ($(IS_LATEST),true)
+	@docker tag $(NAMESPACE)/ezbake:$(version) $(NAMESPACE)/ezbake:latest
+endif
```

Comment:
What's the benefit of tagging as part of `build` as opposed to `publish`?

Reply:
The image is already tagged as part of build, this just adds an additional tag at build time. This is a standard pattern we have for pupperware containers, and tagging at build time is very useful for local investigations/tests